code/model/modelreplace.cpp

```diff
@@ -0,0 +1,409 @@
+				pm->submodel[old_n_submodel].num_children = 0;
+				pm->submodel[old_n_submodel].first_child = -1;
+			}
+
+			//Actually copy textures
+			for (const auto& usedTexture : textureIDReplace) {
+				pm->maps[usedTexture.second] = appendingPM->maps[usedTexture.first];
+				keepTextures.emplace(usedTexture.first);
+			}
+		}
+		else {
+			Warning(LOCATION, "Failed to add submodel to virtual POF, original POF already has a subsystem with the same name as was supposed to be added.");
+		}
+	}
+	else {
+		Warning(LOCATION, "Failed to add submodel to virtual POF, specified subobject was not found. Returning original POF");
+	}
+
+	
+	model_page_out_textures(appendingPM, true, keepTextures);
+	model_free(appendingPM);
+}
+
+VirtualPOFOperationRenameSubobjects::VirtualPOFOperationRenameSubobjects() {
+	while (optional_string("+Replace:")) {
+		SCP_string replace;
+		stuff_string(replace, F_NAME);
+		SCP_tolower(replace);
+		required_string("+With:");
+		stuff_string(replacements[replace], F_NAME);
+	}
+}
+
+void VirtualPOFOperationRenameSubobjects::process(polymodel* pm, model_read_deferred_tasks& deferredTasks, model_parse_depth /*depth*/) const {
+	for (int i = 0; i < pm->n_models; i++) {
+		auto it = replacements.find(pm->submodel[i].name);
+		if (it != replacements.end()) {
+			strncpy(pm->submodel[i].name, it->second.c_str(), it->second.size());
+			pm->submodel[i].name[it->second.size()] = '\0';
+		}
+		it = replacements.find(pm->submodel[i].lod_name);
+		if (it != replacements.end()) {
+			strncpy(pm->submodel[i].lod_name, it->second.c_str(), it->second.size());
+			pm->submodel[i].lod_name[it->second.size()] = '\0';
+		}
+	}
+
+	auto copy_model_subsys = deferredTasks.model_subsystems;
+	for (const auto& replace : replacements) {
+		deferredTasks.model_subsystems.erase(replace.first);
+	}
+	for (const auto& replace : replacements) {
+		auto it = copy_model_subsys.find(replace.first);
+		if (it != copy_model_subsys.end())
+			deferredTasks.model_subsystems.emplace(replace.second, it->second);
+	}
+
+	auto copy_engine_subsys = deferredTasks.engine_subsystems;
+	for (const auto& replace : replacements) {
+		deferredTasks.engine_subsystems.erase(replace.first);
+	}
+	for (const auto& replace : replacements) {
+		auto it = copy_engine_subsys.find(replace.first);
+		if (it != copy_engine_subsys.end())
+			deferredTasks.engine_subsystems.emplace(replace.second, it->second);
+	}
+}
+
+VirtualPOFOperationChangeData::VirtualPOFOperationChangeData() {
+	required_string("+Submodel:");
+	stuff_string(submodel, F_NAME);
+	SCP_tolower(submodel);
+
+	if (optional_string("+Set Offset:")) {
+		setOffset = make_unique<vec3d>();
+		stuff_vec3d(setOffset.get());
+	}
+}
+
+void VirtualPOFOperationChangeData::process(polymodel* pm, model_read_deferred_tasks& deferredTasks, model_parse_depth /*depth*/) const {
+	int subobj_no = -1;
+	for (int i = 0; i < pm->n_models; i++) {
+		if (!stricmp(pm->submodel[i].name, submodel.c_str()))
+			subobj_no = i;
```

Comment:
Here too the comment about breaking out of the loop, or about creating a new function to search for a submodel name